test/panic_style/defs.bzl

```diff
@@ -0,0 +1,29 @@
+"""Test transitions to test panic_style."""
+
+def _panic_style_transition_impl(_settings, attr):
+    return {
+        "//:panic_style": attr.panic_style,
+    }
+
+_panic_style_transition = transition(
+    implementation = _panic_style_transition_impl,
+    inputs = [],
+    outputs = ["//:panic_style"],
+)
+
+def _with_panic_style_cfg_impl(ctx):
+    return [DefaultInfo(files = depset(ctx.files.srcs))]
+
+with_panic_style_cfg = rule(
+    implementation = _with_panic_style_cfg_impl,
+    attrs = {
+        "panic_style": attr.string(mandatory = True),
+        "srcs": attr.label_list(
```

Comment:
This is not really `srcs` but rather `binary`, right? As in, we pass the executable generated by a `rust_binary` here?

Reply:
In this case, yes. If you look at this rule as a more general way of applying that transition to any set of things, then `srcs` is the more common name. I don't really care which name, do you want me to change it?